tls/s2n_handshake_io.c

```diff
@@ -852,6 +852,20 @@ message_type_t s2n_conn_get_current_message_type(struct s2n_connection *conn)
     return ACTIVE_MESSAGE(conn);
 }
 
+static S2N_RESULT s2n_execute_handler(struct s2n_connection *conn)
+{
+    RESULT_ENSURE_REF(conn);
+
+    /* Ensure the state machine referenced is consistant throughout the handshake */
+    if (conn->actual_protocol_version_established) {
```

Comment:
You're setting the flag earlier now, but we're still not checking it until the version is negotiated bc of this if. Is that a problem? The TLS1.3 and TLS1.2 handshakes can vary before the version is negotiated.